refactor: move commands away from recipe

abeatrix opened the pull request:
This PR includes the following changes:
- Moves commands and custom commands away from recipes.
- Addresses issue detailed in https://github.com/sourcegraph/cody/issues/2548
- Removes `SimpleChatRecipeAdapter`
- Removes `executeRecipe` from SimpleChatPanelProvider and ChatManager
- Removes the `custom-prompts` recipe which was used for commands and custom commands
- Runs `executeCommand` from top level in `main.ts` and turns it into a vscode command
- Fixes issue where all context is showing up as `via embeddings`
- Adds range to editor context for commands RE: https://github.com/sourcegraph/cody/issues/1984
- Updates to execute all edit commands in CommandRunner so that we won't need to rely on sidebar view to execute fixup or custom commands recipes.

## Test plan

<!-- Required. See https://sourcegraph.com/docs/dev/background-information/testing_principles. -->

All the current tests for chat, commands, and custom commands should still pass and work correctly.

1. Run the explain command
   - [ ] from sidebar
   - [ ] from left-click menu
   - [ ] from chat
2. Run the doc command
   - [ ] from sidebar
   - [ ] from left-click menu
   - [ ] from chat
3. Run a custom command
   - [ ] from sidebar
   - [ ] from left-click menu
   - [ ] from chat
4. Run an edit command

- Make sure context for each command is showing up correctly

## Demo

https://github.com/sourcegraph/cody/assets/68532117/7e4d9118-9a0a-48cd-823a-af2646e52357

#### Before - context for test command are now showing up in context list

![image](https://github.com/sourcegraph/cody/assets/68532117/9471dcee-cdce-499a-8eae-caf7dd3ce607)


#### After

- Context sources for commands + Context for text command are showing up correctly
- Context for commands include range

![image](https://github.com/sourcegraph/cody/assets/68532117/c95b4407-98c4-47af-a288-76fce10a0d5e)

dominiccooney commented on vscode/test/e2e/at-file-context-selecting.test.ts:
```diff
@@ -89,8 +89,9 @@ test('@-file empty state', async ({ page, sidebar }) => {
         )
     ).toBeVisible()
 
+    // TODO bee fix
```
What's the plan for this? If it's a TODO to land, let's specify what and when.

abeatrix replied:
Woops i forgot to delete this, will update. Thank you!

dominiccooney commented:
It would be great if we added automated tests to cover the essential functionality not already covered by tests.

abeatrix commented on vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:
```diff
@@ -995,9 +995,36 @@ class ContextProvider implements IContextProvider {
+            (await this.editor.getActiveTextEditorSmartSelection()) ||
+            this.editor.getActiveTextEditorSelectionOrVisibleContent()
+        if (!selection || isCodyIgnoredFile(vscode.Uri.file(selection.fileName))) {
+            return []
```
Good questions! We check if it's an ignored file or not when we try to create context item from a selection object.

If a command requires user to have a current selection, we will try to use Smart Selection to auto select code for the user. We won't try to fall back to use visible context as selection for commands that require selection specifically.

If a command doesn't require a selection, we will try to get the visible context to use as selection context, like the explain or smell commands for example, unless the command.context.selection for the command is specified to be false.

Does this answer your questions? If yes, I will add these as docstring!

abeatrix replied:
TL:DR 
- we check for isCodyIgnoredFile  in ContextProvider when creating context item
- We generate error about no file opened in webview provider where we handle command execution

dominiccooney reviewed (approved):
Looks good. Please consider adding automated tests in a follow up change so we can keep broadening the contributor base.

abeatrix commented:
> Looks good. Please consider adding automated tests in a follow up change so we can keep broadening the contributor base.

Updated existing tests and added new tests for the new simple chat provider! Thanks for the review @dominiccooney!
